pkg/handlers/v1/hydrator.go

```diff
@@ -59,6 +60,17 @@ func (h *HydrationHandler) Handle(ctx context.Context, evt AssetEvent) error {
 	}
 	assetVulnEvent := domainAssetVulnerabilityDetailsToEvent(assetWithVulnerabilityDetails)
 	_, err = h.Producer.Produce(ctx, assetVulnEvent)
+	switch err.(type) {
+	case producer.ErrSizeLimitExceeded:
+		logger.Error(logs.PayloadSizeLimitExceededError{
+			Reason:      err.Error(),
+			LastScanned: assetVulnEvent.LastScanned.Format(time.RFC3339Nano),
+			ID:          int(assetVulnEvent.ID),
+			Hostname:    assetVulnEvent.Hostname,
+			IP:          assetVulnEvent.IP,
+		})
+		return nil
```

Comment:
Suggestion: The logline will be crucial in helping us fix the issue, but I think we should still return a failure response from the service. This is still a failure, and fixing the failure is more important than reporting an artificially high success rate for the service.

Reply:
Works for me, will fix.

Reply:
I'm not familiar with the semantics of returning an error, but will that result in this worker picking the work back up and retrying unnecessarily?

Reply:
With the current configuration, yes. We can and should, coerce the handler response into a 400 Bad Request response so that it will be retried _fewer_ times, but this is still a problem that I'm currently working on a solution for across all of our services.